### PR TITLE
Preview ImageVector with group params

### PR DESCRIPTION
**Pull request by egorikftp**
- closes https://github.com/ComposeGears/Valkyrie/issues/225

- Added more tests

Screenshot 2024-10-11 at 14 58 01

### CODE REVIEW

File: components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt [hunk 1]
```diff
@@ -1,5 +1,16 @@
...
 inline fun <reified T : Any> Any?.cast(): T = this as T
+
+inline infix fun <T> T?.or(default: T): T = this ?: default
+
+fun Any?.toStringOrNull(): String? {
+    return when (val value = toString()) {
+        "null" -> null
+        else -> value
+    }
+}
```
Comment (Goooler) [inline components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt:16]:
```suggestion
```

I guess you want `this?.toString()`.

File: components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt [hunk 2]
```diff
@@ -1,5 +1,16 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
 package io.github.composegears.valkyrie.extensions
 
 inline fun <reified T : Any> Any?.safeAs(): T? = this as? T
 
 inline fun <reified T : Any> Any?.cast(): T = this as T
+
+inline infix fun <T> T?.or(default: T): T = this ?: default
```
Comment (Goooler) [inline components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt:9]: This looks redundant if you are using it as an infix.

**Review by Goooler — APPROVED** [review]

File: components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt (hunk 1 above)
Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt:16]: Oh, yes

I tried `nullable.toString()` and it returns "null" 😄

File: components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt (hunk 2 above)
Comment (egorikftp) [reply components/extensions/src/main/kotlin/io/github/composegears/valkyrie/extensions/Any.kt]: agree, reverted

**Pull request merged**